src/frontend/src/utils/iiConnection.ts

```diff
@@ -687,6 +697,6 @@ export const inferHost = (): string => {
     return location.protocol + "//" + location.host;
   }
 
-  // In general, use the official IC HTTP domain.
-  return location.protocol + "//" + IC_API_DOMAIN;
+  // Otherwise assume it's a custom setup and use the host itself as API.
+  return location.protocol + "//" + location.host;
```

Comment:
> I don't know that it is, but I reused [code from agent-js](https://github.com/dfinity/agent-js/blob/main/packages/agent/src/agent/http/index.ts#L205) to be safe

I think the point of that snippet in agent-js is to disallow a change of protocol using `options` (and only require configuration of the host and not the whole origin). But this does not apply here, since both the `protocol` and the `host` come from the same source anyway.

Up to you, how you want to handle it.

Reply:
Probably, though again, I just wanted to not be caught off guard by some weird browser behavior and use something that I knew worked. Really don't care beyond that, though would change this in a separate. PRs welcome!